ui/packages/catalog-ui-search/src/main/webapp/react-component/container/sharing/sharing.tsx

```diff
@@ -110,7 +115,46 @@ export class Sharing extends React.Component<Props, State> {
         values: users.filter(e => e.access === Access.Share).map(e => e.value),
       },
     ]
+    this.attemptSave(attributes)
+  }
+
+  attemptSave = (attributes: any) => {
+    fetch('/search/catalog/internal/metacard/' + this.props.id)
+      .then(res => res.json())
+      .then(data => {
+        // Only allow a user to save if the previous workspace state is the same as the latest workspace state
+        // so that changes made by other users are not clobbered.
+        // NOTE: This is a temporary workaround.
+        if (
+          JSON.stringify(data.metacards[0]) ===
```

Comment:
Do you have access here to the attribute for last modified date? That comparison would be a little faster than the whole metacard.

Reply:
That's a good point. I'll try it out